javascript/javascript2/week1/index.html

```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>spirit animal name generator</title>
+    <script type="text/javascript" src="index.js" defer></script>
```

Comment:
Its a best practice to write the script tag in the end of the body unless you want to load the functionality before the contents is loaded

Reply:
But I watch from one article, he said it is the best way of this way.so I am confused now.

Reply:
It depends on situation. If you want to load some functionality before the content loads you can place the script tag in head section. But normally we do the functionalities after the content of page and css is loaded, so script tag at bottom.

Reply:
Conclusion
The current state-of-the-art is to put scripts in the <head> tag and use the async or defer attributes. This allows your scripts to be downloaded ASAP without blocking your browser.

The good thing is that your website should still load correctly on the 2% of browsers that do not support these attributes while speeding up the other 98%.
https://stackoverflow.com/questions/436411/where-should-i-put-script-tags-in-html-markup